init-tracing-opentelemetry/src/config.rs

```diff
@@ -523,10 +523,13 @@ impl TracingConfig {
         // Build the final subscriber based on OTEL configuration
         if self.otel_config.enabled {
             let subscriber = transform(tracing_subscriber::registry());
-            let (subscriber, otel_guard) = regiter_otel_layers(subscriber)?;
-            let subscriber = subscriber
-                .with(self.build_layer()?)
-                .with(self.build_filter_layer()?);
+            let layer = self.build_layer()?;
+            let filter_layer = self.build_filter_layer()?;
+            let (subscriber, otel_guard) = register_otel_layers_with_resource(
+                subscriber,
+                self.otel_config.resource_config.unwrap_or_default().build(),
+            )?;
+            let subscriber = subscriber.with(layer).with(filter_layer);
```

Comment:
`self.otel_config.resource_config.unwrap_or_default().build()` consumes self partially, so we cannot call `self.build_layer()` or `self.build_filter_layer()` here.